shared/src/main/java/uk/gov/di/authentication/shared/state/StateMachine.java

```diff
@@ -62,18 +62,23 @@ public StateMachine(Map<T, List<Transition<T, A, C>>> states) {
     }
 
     public T transition(T from, A action) {
-        return states.getOrDefault(from, emptyList()).stream()
-                .filter(t -> t.getAction().equals(action))
-                .findFirst()
-                .orElseThrow(
-                        () -> {
-                            LOGGER.info(
-                                    "Session attempted invalid transition from {} on action {}",
-                                    from,
-                                    action);
-                            return new InvalidStateTransitionException();
-                        })
-                .getNextState();
+        T to =
+                states.getOrDefault(from, emptyList()).stream()
+                        .filter(t -> t.getAction().equals(action))
+                        .findFirst()
+                        .orElseThrow(
+                                () -> {
+                                    LOGGER.error(
```

Comment:
Can we make this a method reference to something like `this::logAndThrowException`?

Just a taste thing, inline lambdas push the indention in a lot.